src/Plugin/DataType/LoggerEntry.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\DataType\LoggerEntry.
+ */
+
+namespace Drupal\rules\Plugin\DataType;
+
+use Drupal\Core\TypedData\Plugin\DataType\Map;
+
+/**
+ * The logger entry data type.
+ *
+ * @see: https://www.drupal.org/node/2625238
+ *
+ * @ingroup typed_data
+ *
+ * @DataType(
+ *   id = "loggerentry",
+ *   label = @Translation("Logger Entry"),
+ *   definition_class = "\Drupal\Core\TypedData\MapDataDefinition"
+ * )
+ */
+class LoggerEntry extends Map {
+
+  /**
+   * The data definition.
+   *
+   * @var \Drupal\Core\TypedData\ComplexDataDefinitionInterface
+   */
+  protected $definition;
+
+  /**
+   * An array of values for the contained properties.
+   *
+   * @var array
+   */
+  protected $values = array();
+
+  /**
+   * The array of properties.
+   *
+   * @var \Drupal\Core\TypedData\TypedDataInterface[]
+   */
+  protected $properties = array();
+
+  public function __construct(DataDefinitionInterface $definition, $name, TypedDataInterface $parent, $logger_entry_context) {
```

Comment:
I think the constructor should not take the logger_entry array values, typed data is always set with the setValue() method.